chef/cookbooks/ceph/recipes/conf.rb

```diff
@@ -62,6 +62,9 @@
 # calculate pg_num based on documentation
 # http://ceph.com/docs/master/rados/operations/placement-groups/
 case num_osds
+when 0
+  Chef::Log.warn("Ceph recipe invoked but there are no OSDs!")
+  return
```

Comment:
Ah, good thought. So, if the search somehow erroneously came up with no OSDs on one run, but we let it through with a default pg_num, then the config and presumably some services would still be deployed, but with a slightly bogus configuration. Later, at the next chef-client run, if the search fixed itself, the right config would then be written, correct?